crates/router/src/db/refund.rs

```diff
@@ -770,4 +831,46 @@ impl RefundInterface for MockDb {
             .cloned()
             .collect::<Vec<_>>())
     }
+
+    #[cfg(feature = "olap")]
+    async fn filter_refund_by_meta_constraints(
+        &self,
+        _merchant_id: &str,
+        _refund_details: &api_models::refunds::RefundListRequest,
+        _storage_scheme: enums::MerchantStorageScheme,
+        _limit: i64,
+        _offset: i64,
+    ) -> CustomResult<api_models::refunds::RefundListMetaData, errors::StorageError> {
+        let mut refund_meta_data = api_models::refunds::RefundListMetaData {
+            connector: vec![],
+            currency: vec![],
+            status: vec![],
+        };
+
+        let refunds = self.refunds.lock().await;
+
+        for refund in refunds.clone().into_iter() {
```

Comment:
Instead of checking for the presence of an element in the `Vec`, consider collecting them all to a `HashSet` instead.